pkg/agent/route/route_windows.go

```diff
@@ -364,8 +442,14 @@ func (c *Client) listRoutes() (map[string]*util.Route, error) {
 		if !rt.DestinationSubnet.IP.IsGlobalUnicast() {
 			continue
 		}
-		// Windows adds an active route entry for the local broadcast address automatically when a new IP address
-		// is configured on the interface. This route entry should be ignored in the list.
+		// When configuring an IP address to an interface on Windows, three route entries will be automatically added.
+		// For example, if the IP address is 10.10.0.1/24, the following three routes will be created:
+		// Network Destination   Netmask          Gateway  Interface  Metric
+		// 10.10.0.1             255.255.255.255  On-link  10.10.0.1  281
+		// 10.10.0.0             255.255.255.0    On-link  10.10.0.1  281
+		// 10.10.0.255           255.255.255.255  On-link  10.10.0.1  281
+		// The host (10.10.0.1) and broadcast (10.10.0.255) routes should be ignored since they are unnecessary and redundant.
+		// We can ignore them by comparing them to the calculated broadcast IP.
 		if !rt.GatewayAddress.Equal(config.VirtualServiceIPv4) && rt.DestinationSubnet.IP.Equal(iputil.GetLocalBroadcastIP(rt.DestinationSubnet)) {
```

Comment:
The host route is ignored here since the broadcast IP of the destination IP in host route is the destination itself. @tnqn

Reply:
then should we remove the new check added here https://github.com/antrea-io/antrea/pull/3889/files/720023a9ed444a790842d8a62db295f70161d315#r1153399542? otherwise the code is contradictory.